modulus/models/module.py

```diff
@@ -57,17 +57,32 @@ class Module(torch.nn.Module):
 
     def __new__(cls, *args, **kwargs):
         out = super().__new__(cls)
+
+        # Get signature of __init__ function
         sig = inspect.signature(cls.__init__)
+
+        # Bind args and kwargs to signature
         bound_args = sig.bind_partial(
             *([None] + list(args)), **kwargs
         )  # Add None to account for self
         bound_args.apply_defaults()
-        bound_args.arguments.pop("self", None)
 
+        # Get args and kwargs (excluding self and unroll kwargs)
+        instantiate_args = {}
+        for param, (k, v) in zip(sig.parameters.values(), bound_args.arguments.items()):
+            if k != "self":  # Skip self
+                if param.kind == param.VAR_KEYWORD:
+                    instantiate_args.update(v)
```

Comment:
This is a nice solution! There is no need for a recursive approach here, is there? Like if this dict itself had a dict entry, etc?

Reply:
Apologies but I don't think I understand your comment. Was the issue with the if statements? Not sure I see how this is a recursive approach.

Reply:
I was just asking if there was a need for a recursive approach here. Sounds like no.